src/databricks/labs/ucx/source_code/base.py

```diff
@@ -15,8 +15,14 @@
 
 @dataclass
 class Advice:
+
+    MISSING_SOURCE_TYPE = "<MISSING_SOURCE_TYPE>"
+    MISSING_SOURCE_PATH = "<MISSING_SOURCE_PATH>"
+
     code: str
     message: str
+    source_type: str
+    source_path: str
```

Comment:
we should keep WHAT (e.g. deprecation) problem we have from WHERE (e.g. notebook or a file) that problem is. 

```
@dataclass
class NotebookAdvice:
  path: str
  advice: list[Advice]

@dataclass
class FileAdvice:
  path: str
  advice: list[Advice]
```

that being said, we may need another datastructure to map onto a database table, but it should have Job ID and Job Name as well and most likely live within the `assessment` package:

```
@dataclass
class WorkloadAdvice:
  workflow_id: int
  workflow_name: str
  object_type: string
  object_path: string # alternatively, object_id, to make links easier
  code: str
  message: str
  line: int
```

Reply:
See my comment in #1448